### DISCUSSION TITLE
Zig Programlama Dili Modern, Güvenli ve Esnek | SonsuzUs Github

### DISCUSSION THREAD

**Discussion by giscus** [sonsuzus/sonsuzus.github.io#361] [Site]
# Zig Programlama Dili Modern, Güvenli ve Esnek | SonsuzUs Github

Zig, Andrew Kelley tarafından geliştirilen, düşük seviyeli ve genel amaçlı bir programlama dilidir. Zig, C diline benzer bir söz dizimine sahip olmakla birlikte, daha modern ve güvenli özelliklere odaklanmıştır. Amaçları arasında hız, düşük seviyeli programlama ve bellek güvenliği bulunur.

https://sonsuzus.github.io/posts/zig-programlama/

**Comment by sonsuzus** [comment]
```zig
const std = @import("std");  

fn is_prime(n: u32) bool {  
    if (n < 2) return false;  
    for (i: u32 = 2; i * i <= n; i += 1) {  
        if (n % i == 0) return false;  
    }  
    return true;  
}  

pub fn main() void {  
    const out = std.io.getStdOut().writer();  
    
    for (i: u32 = 2; i <= 1000; i += 1) {  
        if (is_prime(i)) {  
            const result = std.fmt.print(out, "{d}\n", i);  
            if (result.isError()) {  
                std.debug.print("Error writing to output.\n", .{});  
                return;  
            }  
        }  
    }  
}
```

is_prime Fonksiyonu: Bu fonksiyon, verilen bir sayının asal olup olmadığını kontrol eder. Sayı 2'den küçükse false döner. Daha sonra, i 2’den başlayarak n’in kareköküne kadar olan sayılarla n’nin bölenlerini kontrol eder.

main Fonksiyonu: 2 ile 1000 arasındaki her sayıyı döngü ile kontrol eder. Eğer is_prime fonksiyonu true dönerse, o sayıyı standart çıktıya yazdırır.
